testJS/secret.html

```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Dummy HTML with API Keys and Passwords</title>
+    <style>
+        body {
+            font-family: Arial, sans-serif;
+        }
+        .warning {
+            background-color: #f8d7da;
+            padding: 10px;
+            border: 1px solid #f5c6cb;
+            color: #721c24;
+        }
+        .hidden-secrets {
+            display: none;
+        }
+    </style>
+</head>
+<body>
+
+    <h1>Dummy HTML File Containing Secrets</h1>
+    <p>This is an example HTML file containing dummy secrets for educational purposes.</p>
+
+    <!-- SECRET: API KEY -->
+    <script>
+        // API Key (don't do this in production!)
+        const apiKey = "sk_test_4eC39HqLyjWDarjtT1zdp7dc"; // Example API Key
+
+        // Example usage of the API key
+        console.log("Using API key:", apiKey);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider removing sensitive data logging</b></div>
<div id="fix">

Logging sensitive information like API keys, credentials and private keys to the console creates security risks as these logs may be exposed in browser dev tools or log files.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -34,1 +34,1 @@
 -        console.log("Using API key:", apiKey);
 +        console.log("Using API key: [REDACTED]");

 @@ -44,2 +44,1 @@
 -        console.log("Connecting to database with username:", dbUsername);
 -        console.log("Password:", dbPassword);
 +        console.log("Connecting to database with username: [REDACTED]");

 @@ -54,1 +54,1 @@
 -        console.log("Using private key:", privateKey);
 +        console.log("Using private key: [REDACTED]");
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PratikDash123/CRA_Automation/pull/4#issuecomment-2510777966>#5d9957</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged